dandi-info/config.yaml.j2

```diff
@@ -27,24 +29,51 @@ hub:
       import os
       import warnings
 
-      from oauthenticator.github import GitHubOAuthenticator
+      from oauthenticator.github import GitHubOAuthenticator, _api_headers
+      from tornado.httpclient import HTTPRequest
 
       # define our OAuthenticator with `.pre_spawn_start`
       # for passing auth_state into the user environment
 
       class GitHubEnvAuthenticator(GitHubOAuthenticator):
 
+          async def authenticate(self, handler, data=None):
```

Comment:
i had to override authenticate and not post_auth_hook. the latter was already too late.

Reply:
Ah crap okay. I'll want to open an issue in JupyterHub about this i think, asking if it can make sense to add some pre-authorization hook.